SoObjects/SOGo/SOGoUserManager.m

```diff
@@ -372,10 +373,22 @@ - (NSString *) getExternalLoginForUID: (NSString *) uid
         dd = [SOGoDomainDefaults defaultsForDomain: domain];
       else
         dd = [SOGoSystemDefaults sharedSystemDefaults];
-      
-      login = [dd forceExternalLoginWithEmail] ? [self getEmailForUID: uid] : uid;
+
+      if ([dd forceExternalLoginWithEmail])
+        {
+          sd = [SOGoSystemDefaults sharedSystemDefaults];
+          if ([sd enableDomainBasedUID])
+            {
+              // On multidomain environment we must use uid@domain
+              // for getEmailForUID method
+              login = [NSString stringWithFormat: @"%@@%@", uid, domain];
```

Comment:
It's impossible at this point that `uid` has the domain (and if it does, there is another bug or design problem elsewhere). This method is called using `loginInDomain` as input parameter which should return the username part.

Personally I prefer to not include a check that could mask another issue.